### DISCUSSION TITLE
Can I use this Code with an ESP8266 (Arduino)?

### DISCUSSION THREAD

**Discussion by jannis-weber** [googleapis/google-cloud-cpp#6817] [Q&A] [answered]
Hello!
I'm searching despertaly for a library to use with my ESP8266. Is it possible to somehow convert that code to arduino-cpp or something like that? If not, do you know a working library to connect my esp to Pub/Sub?
Thank you guys!

**Comment by coryan** [comment, marked as answer]
I am afraid we don't have enough enough experience with arduino to even provide you with a good answer.  This might help: to use Cloud Pub/Sub with `google-cloud-cpp` you are going to need a number of dependencies to work, including: [Abseil](https://gitbhub.com/abseil/abseil-cpp), [protobuf](https://github.com/protocolbuffers/protobuf), and [gRPC](https://github.com/grpc/grpc).

Both protobuf and gRPC include _core generators_, which implies you need to cross-compile them so you can generate code in the host environment (your workstation), but the generated code must be compiled for the target environment (arduino).
